sky/data/storage.py

```diff
@@ -1865,10 +1902,615 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+            self.resource_group_name = data_utils.get_az_resource_group(
+                self.storage_account_name)
+        # Creates new resource group and storage account or use the 
+        # storage_account provided by the user through config.yaml
+        else:
+            config_storage_account = skypilot_config.get_nested(
+                ('azure', 'storage_account'), None)
+            if config_storage_account is not None:
+                # using user provided storage account from config.yaml
+                self.storage_account_name = config_storage_account
+                self.resource_group_name = data_utils.get_az_resource_group(
+                    self.storage_account_name)
+                # when the provided storage account does not exist under user's
+                # subscription id.
+                if self.resource_group_name is None:
+                    with ux_utils.print_exception_no_traceback():
+                        raise ValueError(
+                            'The storage account '
+                            f'{self.storage_account_name!r} specified in '
+                            'config.yaml does not exist under the user\'s '
+                            'subscription ID. Provide a storage account '
+                            'through config.yaml only when creating a '
+                            'container under an already existing storage '
+                            'account within your subscription ID.'
+                        )
+            else:
+                # If storage account name is not provided from config, then 
+                # use default resource group and storage account names.
+                self.storage_account_name = f'sky{self.region}{common_utils.get_user_hash()}'
+                self.resource_group_name = f'sky{common_utils.get_user_hash()}'
+                self._init_storage_account_and_resource_group()
+
+        # resource_group_name is set to None when using non-sky-managed
+        # public container or private container without authorization.
+        if self.resource_group_name is not None:
+            self.storage_account_key = data_utils.get_az_storage_account_key(
+                self.storage_account_name, self.resource_group_name,
+                self.storage_client, self.resource_client)
+
+        self.bucket_name, is_new_bucket = self._get_bucket()
+        if self.is_sky_managed is None:
+            # If is_sky_managed is not specified, then this is a new storage
+            # object (i.e., did not exist in global_user_state) and we should
+            # set the is_sky_managed property.
+            # If is_sky_managed is specified, then we take no action.
+            self.is_sky_managed = is_new_bucket
+
+    def _init_storage_account_and_resource_group(self) -> None:
+        """Initialize storage account and resource group.
+
+        Checks if the set resource group name and storage account name already
```

Comment:
the resources group name cannot be set now? pls update comment

Reply:
This is outdated function. Rewritten as `_get_storage_account_and_resource_group` with updated comment.